src/pages/Search/AdvancedSearchFilters.tsx

```diff
@@ -136,81 +136,104 @@ function AdvancedSearchFilters() {
     const taxRates = getAllTaxRates();
     const personalDetails = usePersonalDetails();
 
+    const currentType = searchAdvancedFilters.type ?? CONST.SEARCH.DATA_TYPES.EXPENSE;
+    const shouldShowExpenseSpecificFilter = currentType === CONST.SEARCH.DATA_TYPES.EXPENSE;
+    const shouldShowChatSpecificFilter = currentType === CONST.SEARCH.DATA_TYPES.CHAT;
```

Comment:
I think we could declare different arrays for each type and then return the values depending on type, similar to what we do [here](https://github.com/Expensify/App/blob/77ed3e1e22d559de736178268ed036a0396bee0f/src/components/Search/SearchStatusBar.tsx#L110-L120)

Reply:
@luacmartins could you create a short list of type-specific filters and indicate when they should be visible? (e.g., `Has` is only visible when type=chat). It would be a huge help.

Reply:
I think this is the correct list. @JmillsExpensify could you please confirm?

**Expense**
Date, currency, merchant, description, reportID, total, category, keywords, taxRate, expenseType, tag, from, to

**Invoice**
Date, currency, merchant, description, reportID, total, category, keywords, taxRate, tag, from, to

**Trip**
Date, currency, merchant, description, reportID, total, category, keywords, taxRate, tag, from, to

**Chat**
Date, keywords, from, to, in, has, is

Reply:
Assuming we want to confirm the full list, aren't we missing `card`? Then for `total` did we decide to make that context-specific or will we have distinct filters for `expense-total` and `report-total`?

I also think we should have `card` for invoice and trip given that those expenses could have been made on a credit card, right? 

Finally comment is `to` for chats. This is who you sent any message to? That could get a little confusing, as I guess it's implied that it's from "you" and you can change that if you want? I'm not convinced we need that filter to start.

Reply:
>Assuming we want to confirm the full list, aren't we missing card?

Ah yes, I missed the card filter. 

>Then for total did we decide to make that context-specific or will we have distinct filters for expense-total and report-total?

This is what we have in the doc:

The syntax is context specific, so if you are searching for a total on the Expenses filter, you’re searching for expense totals. If you are searching for a total on report-focused filter (e.g. Drafts), then you’re searching for report totals. 

>This is who you sent any message to?

Correct. We can remove the filter from chats. 

So the final list is:

**Expense**
Date, currency, merchant, description, reportID, total, category, keywords, taxRate, expenseType, tag, from, to, cardID

**Invoice**
Date, currency, merchant, description, reportID, total, category, keywords, taxRate, tag, from, to, cardID

**Trip**
Date, currency, merchant, description, reportID, total, category, keywords, taxRate, tag, from, to, cardID

**Chat**
Date, keywords, from, in, has, is


Does that look right?